src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -683,12 +601,14 @@ String getFilePath(Calendar calendar, String clientPrefix) {
    *
    * @param blobData list of channels that belongs to the blob
    */
-  void invalidateAllChannelsInBlob(List<List<ChannelData>> blobData) {
+  void invalidateAllChannelsInBlob(List<List<ChannelData<T>>> blobData) {
     blobData.forEach(
         chunkData ->
             chunkData.forEach(
                 channelData -> {
-                  channelData.getVectors().close();
+                  if (channelData.getVectors() instanceof VectorSchemaRoot) {
+                    ((VectorSchemaRoot) channelData.getVectors()).close();
+                  }
```

Comment:
This will be replace by a channelData.close function in the future?

Reply:
ChannelData could be indeed refactored to be polymorphic in the future.
or T could be bound by Closable interface. it can be a good follow-up, especially if there are more use cases for this later